libraries/pbrlib/pbrlib_defs.mtlx

```diff
@@ -321,4 +321,14 @@
     <input name="tangent" type="vector3" defaultgeomprop="Tworld"/>
   </nodedef>
 
+  <!--
+    Node: <fresnel>
+  -->
+  <nodedef name="ND_fresnel_ior" node="fresnel" type="float" nodegroup="pbr"
```

Comment:
Naming the nodedef ND_fresnel_ior, since there might be more fresnel versions added later, for complex ior etc. The node is still just named <fresnel>

Reply:
Hope this is not a dumb question but curious what combinations / connections will be possible.
- different fresnel computations with different types for input / output.
- conversion from complex_ior to artistic_ior (keeping ior always a float)
- call fresnel() with different wavelengths (e.g.) 
Some combination of these / none of these?

Reply:
This fresnel node is mainly usable to do view dependent blending effect in a graph, in the texturing domain. So as inputs to BSDF's or for simple mixing of BSDF's.

However it's not ment as a replacement for the _correct_ fresnel calculations performed inside a BSDF. Inside a BSDF it's done on the microfacet normal sampled from the microfacet distribution, where as in this case on the outside we only have the surface normal and viewdirection to work with. Which means this is only correct for mirror surfaces and not rough surfaces.

However it can be used to _approximate_ the reflectance from a BSDF (ignoring the roughness) which is what I'm using here to weight the emission when placed under the coat in Standard Surface. So as the coat IOR is increased its reflectance is increased and the underlying emission is decreased. An alternative to this would be to add a `reflectance` or `directional_albedo` node that could return the true reflectance for a give BSDF and direction vectors, but this is much harder to implement since it often requires integrating the BSDF lobe, and it's different for different BSDF's. So I decided to settle for this simple approximation instead.

About variations: this one is for dielectric fresnel with a single scalar IOR. We could add one for conductor fresnel as well, which has a complex IOR (it's not as useful since conductors/metals  can't be transmissive like dielectrics so you can't physically layer it on top of something in the same way). Then there are other variations, e.g. where reflectance at facing (or both facing and gracing) angles are given instead of IOR. The variation added here is the most useful one though and solved my emission layering problem, so I think we should wait with any others for now.